Sources/Testing/Events/Recorder/Event.AdvancedConsoleOutputRecorder.swift

```diff
@@ -0,0 +1,79 @@
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2025 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+//
+
+extension Event {
+  /// An experimental console output recorder that provides enhanced test result
+  /// display capabilities.
+  ///
+  /// This recorder is currently experimental and must be enabled via the
+  /// `SWT_ENABLE_EXPERIMENTAL_CONSOLE_OUTPUT` environment variable.
+  ///
+  /// Future capabilities will include:
+  /// - Hierarchical test result display with tree visualization
+  /// - Live progress indicators during test execution
+  /// - Enhanced SF Symbols integration on supported platforms
+  struct AdvancedConsoleOutputRecorder: Sendable {
+    /// Configuration options for the advanced console output recorder.
+    struct Options: Sendable {
+      /// Base console output recorder options to inherit from.
+      var base: Event.ConsoleOutputRecorder.Options
+      
+      /// Whether to enable experimental hierarchical output display.
+      /// Currently unused - reserved for future PR #2.
+      var useHierarchicalOutput: Bool
```

Comment:
There's no need to add these flags if they do nothing. When hierarchical output is added, add the flag at that point.